diff --git a/drivers/video/rockchip/rga2/rga2_mmu_info.c b/drivers/video/rockchip/rga2/rga2_mmu_info.c
index 8fc83cea2701..89bfae6d6151 100644
--- a/drivers/video/rockchip/rga2/rga2_mmu_info.c
+++ b/drivers/video/rockchip/rga2/rga2_mmu_info.c
@@ -312,7 +312,7 @@ static int rga2_MapUserMemory(struct page **pages, uint32_t *pageTable,
 	down_read(&current->mm->mmap_sem);
 #if LINUX_VERSION_CODE < KERNEL_VERSION(4, 6, 0)
 	result = get_user_pages(current, current->mm, Memory << PAGE_SHIFT,
-				pageCount, writeFlag, 0, pages, NULL);
+				pageCount, writeFlag ? FOLL_WRITE:0, pages, NULL);
 #else
 	result = get_user_pages_remote(current, current->mm,
 				       Memory << PAGE_SHIFT,
