--- linux-custom/drivers/gpu/arm/midgard/mali_kbase_mem.c.org	2019-03-07 21:40:22.000000000 +0800
+++ linux-custom/drivers/gpu/arm/midgard/mali_kbase_mem.c	2019-03-10 12:23:52.375281497 +0800
@@ -2182,8 +2182,8 @@ static int kbase_jd_user_buf_map(struct
 	pinned_pages = get_user_pages(NULL, mm,
 			address,
 			alloc->imported.user_buf.nr_pages,
-			reg->flags & KBASE_REG_GPU_WR,
-			0, pages, NULL);
+			reg->flags & KBASE_REG_GPU_WR ? FOLL_WRITE : 0,
+			pages, NULL);
 #elif LINUX_VERSION_CODE < KERNEL_VERSION(4, 9, 0)
 	pinned_pages = get_user_pages_remote(NULL, mm,
 			address,
--- linux-custom/drivers/gpu/arm/midgard/mali_kbase_mem_linux.c.org	2019-03-07 21:40:22.000000000 +0800
+++ linux-custom/drivers/gpu/arm/midgard/mali_kbase_mem_linux.c	2019-03-10 12:27:01.041408283 +0800
@@ -1161,7 +1161,7 @@ static struct kbase_va_region *kbase_mem
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(4, 6, 0)
 	faulted_pages = get_user_pages(current, current->mm, address, *va_pages,
-			reg->flags & KBASE_REG_GPU_WR, 0, pages, NULL);
+			reg->flags & KBASE_REG_GPU_WR ? FOLL_WRITE : 0, pages, NULL);
 #elif LINUX_VERSION_CODE < KERNEL_VERSION(4, 9, 0)
 	faulted_pages = get_user_pages(address, *va_pages,
 			reg->flags & KBASE_REG_GPU_WR, 0, pages, NULL);
